### PR TITLE
Rename state write -> reversible write

### PR DESCRIPTION
**Pull request by z2trillion**
Change these names to be consistent with the code changed in https://github.com/appliedzkp/zkevm-circuits/pull/409

### CODE REVIEW

**Comment by z2trillion** [comment]
@ed255

File: src/SUMMARY.md [hunk 1]
```diff
@@ -15,5 +15,5 @@
     - [Random Linear Combination](./design/random-linear-combinaion.md)
         - [Full Runnable Code](./design/random-linear-combinaion/full-runnable-code.md)
     - [Recursion](./design/recursion.md)
-    - [State Write Reversion Note 1](./design/state-write-reversion.md)
-    - [State Write Reversion Note 2](./design/state-write-reversion2.md)
+    - [Reversible Write Reversion Note 1](./design/reversible-write-reversion.md)
+    - [Reversible Write Reversion Note 2](./design/reversible-write-reversion2.md)
```
Comment (han0110) [inline src/SUMMARY.md:19]: I'm thinking perhaps "Reversion Note" is easier to recognize? Tho I believe people can absolutely tell it's about reversion from this title.

**Review by han0110 — APPROVED** [review]
LGTM! I will also submit a PR to update the text in image later.

File: src/SUMMARY.md (hunk 1 above)
Comment (ed255) [reply src/SUMMARY.md:19]: There are other parts of the documentation that currently use the term "Reversible write reversion" when explaining how reversion is handled, and that seems fine to me.  So I think it's OK to leave this as is.

**Pull request merged**